### PR TITLE
Apollo init

### PR DESCRIPTION
**Pull request by Bazillius**
Based on Prisma brunch I made changes:
- remove Prisma server from the chains of servers
- add simple Apollo express API with knex migration
- change docker-compose file

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
+{
```
Comment (vitaliimelnychuk) [inline tsconfig.json:1]: how about using the same as we have here https://github.com/WiseEngineering/crispy-cli/blob/master/tsconfig.json
especially, for beginning?

File: knexfile.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import * as knex from "knex";
+require('dotenv').config();
```
Comment (vitaliimelnychuk) [inline knexfile.ts:2]: this is good start 👍 

Thinking about moving all configuration into one file, where we will define types
will be something like 
https://github.com/WiseEngineering/crispy-cli/blob/master/src/config.ts
but, there is no need for user config as we will use environment variables for these purposes

Comment (vitaliimelnychuk) [reply knexfile.ts:2]: it's not a blocker here, will add the task for it

File: migrations/20191218152204_users.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import * as Knex from "knex";
+
+
+export async function up(Knex: Knex): Promise<any> {
+    await Knex.schema.createTable('users', table => {
```
Comment (vitaliimelnychuk) [inline migrations/20191218152204_users.ts:5]: not sure we need user schema for start. 
We've decided to start from migrations and them move forward with them 
* logging
* workers
* migrations
* running  flow

We have to look at those areas first
I think we have to change this schema with the `workers` schema. This is the one we really need.
 The interface could simple for start:

```
name: 'beta' //worker name will be sent from server where this migration going to be run
created_at
updated_at
endpoint // where worker is hosted. We will call this API to communicate with worker
api_key? // not required will figure out later how we are going to make auth
```

Let's change this schema with those fields, then we will be ready to use it in client side

File: package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  "version": "1.0.0",
+  "main": "index.js",
+  "repository": "git@github.com:WiseEngineering/crispy-api.git",
+  "author": "Vasyl Zdyrko",
+  "license": "MIT",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "ts-node src/index.ts",
+    "migration": "knex migrate:latest",
+    "precommit": "eslint src/**/*.ts "
```
Comment (vitaliimelnychuk) [inline package.json]:
```suggestion
    "lint": "eslint src/**/*.ts "
```

File: package.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  "name": "crispy-api",
+  "version": "1.0.0",
+  "main": "index.js",
+  "repository": "git@github.com:WiseEngineering/crispy-api.git",
+  "author": "Vasyl Zdyrko",
+  "license": "MIT",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "ts-node src/index.ts",
+    "migration": "knex migrate:latest",
```
Comment (vitaliimelnychuk) [inline package.json]:
```suggestion
    "migrate": "knex migrate:latest",
```

File: package.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "crispy-api",
+  "version": "1.0.0",
+  "main": "index.js",
+  "repository": "git@github.com:WiseEngineering/crispy-api.git",
+  "author": "Vasyl Zdyrko",
```
Comment (vitaliimelnychuk) [inline package.json]:
```suggestion
  "author": "WiseEngineering open-source@wise-engineering.com",
```

File: package.json [hunk 7]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "crispy-api",
+  "version": "1.0.0",
```
Comment (vitaliimelnychuk) [inline package.json]:
```suggestion
  "version": "0.0.1",
```

File: package.json [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    "precommit": "eslint src/**/*.ts "
+  },
+  "dependencies": {
+    "apollo-server-express": "^2.9.14",
+    "express": "^4.17.1",
+    "graphql": "^14.5.8",
+    "graphql-import": "^0.7.1",
+    "knex": "^0.20.4",
+    "mysql": "^2.17.1",
+    "uuid": "^3.3.3"
```
Comment (vitaliimelnychuk) [inline package.json]: is this library in use?

Comment (vitaliimelnychuk) [inline package.json]: let's specify versions in strict mode without `^` to be safe

File: package.json [hunk 9]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "crispy-api",
+  "version": "1.0.0",
+  "main": "index.js",
+  "repository": "git@github.com:WiseEngineering/crispy-api.git",
+  "author": "Vasyl Zdyrko",
+  "license": "MIT",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "ts-node src/index.ts",
```
Comment (vitaliimelnychuk) [inline package.json:10]: not a blocker for now, but need to think about commands for dev/prod server start 

we could even use something like this for dev
`nodemon --watch 'src/**/*.ts' --ext 'ts' --exec 'yarn start'`

File: src/index.ts [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+import { resolvers } from "./graphql";
+
+const typeDefs = importSchema('src/graphql/schema.graphql');
+const server = new ApolloServer({ typeDefs, resolvers });
+const app = express();
+
+server.applyMiddleware({ app });
+app.use('/', express.static('public'));
+
+const port = process.env.SERVER_PORT || 4000;
```
Comment (vitaliimelnychuk) [inline src/index.ts:13]: worth to move it in config file, once we will have one

File: Dockerfile [hunk 11]
```diff
@@ -0,0 +1,13 @@
+FROM node:12.13.1
+
+WORKDIR /code
+
+COPY package.json /code/package.json
+COPY yarn.lock /code/yarn.lock
+RUN yarn install
+
+COPY . /code
```
Comment (vitaliimelnychuk) [inline Dockerfile]: minor comment:
can be changed with docker multistage
https://docs.docker.com/develop/develop-images/multistage-build/

Comment (vitaliimelnychuk) [reply Dockerfile]: To avoid COPY command in dev environment

File: README.md [hunk 12]
```diff
@@ -1,2 +1,42 @@
 # crispy-api
 Typescript GraphQL API for `crispy-web`
+
+API based on [`Apollo`](https://www.apollographql.com/docs/apollo-server/) express server. 
+
+## How to use
```
Comment (vitaliimelnychuk) [inline README.md:6]: Need to specify somewhere that this is developers guide to setup dev environment

We also need a root do user docs

File: docker-compose.yml [hunk 13]
```diff
@@ -0,0 +1,17 @@
+version: '3'
+
+services:
+  mysql:
+    image: mysql:5.7
+    env_file: .env
+    volumes:
+      - ./mysql:/var/lib/mysql
+  graphql:
```
Comment (melnychukvitaliy) [inline docker-compose.yml]:
```suggestion
  crispy-api:
```

File: README.md [hunk 14]
```diff
@@ -1,2 +1,42 @@
...
+Command to start up server
+
+```
+docker-compose up -d
+```
+
+Run migration:
+
+```
+docker exec -it <container name> yarn migration
```
Comment (vitaliimelnychuk) [inline README.md]: maybe worth to specify names in docker-compose to be able  put command here like
```
docker exec -it  crispy-api yarn migrate
```

**Comment by vitaliimelnychuk** [comment]
Issue: #1

File: package.json (hunk 8 above)
Comment (Bazillius) [reply package.json]: nope, will remove

**Review by vitaliimelnychuk — APPROVED** [review]

**Comment by Bazillius** [comment]
@melnychukvitaliy if everything else is ok we can merge this into master

**Comment by vitaliimelnychuk** [comment]
🍑

**Pull request merged**